include/zephyr/net/phy.h

```diff
@@ -98,6 +101,26 @@ __subsystem struct ethphy_driver_api {
  * @endcond
  */
 
+/**
+ * @brief      Soft reset PHY device
+ *
+ * This route performs a PHY soft reset.
+ *
+ * @param[in]  dev     PHY device
+ *
+ * @retval 0 If successful.
+ * @retval -EIO If communication with PHY failed.
+ */
+__syscall int phy_soft_reset(const struct device *dev);
+
+static inline int z_impl_phy_soft_reset(const struct device *dev)
```

Comment:
I noticed that other phy functions are also marked as syscalls which is strange. Any idea why these functions should be able to be called from userspace?

Reply:
No, I just tried to be consistent with the other API functions, I think it doesn't make that much sense indeed.

Reply:
I will send a PR that fixes it.

Reply:
Hmm, I found the original PR https://github.com/zephyrproject-rtos/zephyr/pull/34718. I was questioning the syscall thing at that point too but the change got merged. Dunno if phy API is really working from user space, I can try to do some experiments about it.

Reply:
I do not seem to have suitable Ethernet hw that is using the phy API. @pdgendt if you have time, could you try to call these functions from userspace and report if they really work as expected.

Reply:
I've got an `mixmrt1064_evk`, is there a sample to test?

Reply:
Hah, never mind. I managed to find Atmel e70 board that uses the phy API. I will report how it works with usermode.

Reply:
Yeah, as suspected it did not work in user mode. I might be doing things wrongly but my test app is killed immediately when trying to access the phy device.